.github/workflows/ci.yml

```diff
@@ -69,8 +69,6 @@ jobs:
       - uses: ./.github/actions/setup
         with:
           java-version: ${{ matrix.java-version }}
-      - name: Download all Trino dependencies to make sure cache is complete
-        run: ./.github/bin/download-maven-dependencies.sh
```

Comment:
Why is this part of `Add Maven --offline flag as a default option` commit?
(in fact, I don't know why we have this change at all)

Reply:
We need this in the `setup` action because after it downloads the Maven cache we also need to download any missing dependencies since the cache may be stale or from the target/`main` branch.
If we want to use `--offline` Maven mode each job must first download all the Maven dependencies it requires, and the `setup` job is meant to do just that.